utils/starknet/verifySignature.ts

```diff
@@ -1,57 +1,155 @@
-import { log } from "../discord/logs";
-
+import WatchTowerLogger from "../../watchTower";
 import { callContract } from "./call";
+import { Provider, num, CallData } from "starknet";
+import { ec as EllipticEC } from "elliptic";
 
 type SignatureVerify = {
   signatureValid: boolean;
   error?: string;
 };
 
-export const verifySignature = async (
+export async function isAccountDeployed(
   accountAddress: string,
-  hexHash: string,
-  signature: string[],
   starknetNetwork: string
-): Promise<SignatureVerify> => {
-  // We can verify this message hash against the signature generated in the frontend
-  // by calling the is_valid_signature method on the Account contract
+): Promise<boolean> {
+  const provider = new Provider({
+    sequencer: {
+      baseUrl:
+        starknetNetwork === "mainnet"
+          ? "https://alpha-mainnet.starknet.io"
+          : "https://alpha4-2.starknet.io",
+    },
+  });
+  try {
+    const code = await provider.getCode(accountAddress);
+    return code.bytecode.length > 0;
+  } catch (error) {
+    WatchTowerLogger.error(
+      `Failed to check deployment status for ${accountAddress}: ${error}`
+    );
+    return false;
+  }
+}
+
+export function starknetEcVerify(
+  pubkey: string,
+  messageHash: string,
+  signature: string[]
+): boolean {
   try {
-    const result = await callContract({
-      starknetNetwork:
-        starknetNetwork === "mainnet" ? "mainnet" : ("sepolia" as any),
-      contractAddress: accountAddress,
-      entrypoint: "isValidSignature",
-      calldata: [hexHash, `${signature.length}`, ...signature],
+    const curve = new EllipticEC("secp256k1");
+    const key = curve.keyFromPublic(pubkey, "hex");
+    const msgHashBigInt = num.toBigInt(messageHash);
+    const [r, s] = signature.map((sig) => num.toBigInt(sig));
+
+    const isValid = key.verify(num.toHex(msgHashBigInt), {
+      r: num.toHex(r),
+      s: num.toHex(s),
     });
 
-    const signatureValid = result[0] === "0x1";
+    return isValid;
+  } catch (error) {
+    WatchTowerLogger.error(
+      `ECDSA verification failed for pubkey ${pubkey}: ${error}`
+    );
+    return false;
+  }
+}
 
-    return {
-      signatureValid,
-      error: signatureValid ? undefined : `Invalid signature ${result[0]}`,
-    };
-  } catch (e: any) {
+export const verifySignature = async (
+  accountAddress: string,
+  hexHash: string,
+  signature: string[],
+  starknetNetwork: string,
+  pubkey?: string
```

Comment:
I'm pretty sure that the pubkey and the accountAddress are exactly the same thing